src/Classes/ServiceAPI/MyRadio_Creditable.php

```diff
@@ -108,16 +108,19 @@ public function getPresenterString()
                 $credit_types_in_byline[] = $type["value"];
             }
         }
-        $str = '';
+        $names = [];
         foreach ($this->getCredits() as $credit) {
             if (in_array($credit['type'], $credit_types_in_byline)) {
-                $str .= $credit['User']->getName().', ';
+                $names[] = $credit['User']->getName();
             } else {
                 continue;
             }
         }
+        $names = array_unique($names);
+        $str = implode(", ",$names);
+        
 
-        return empty($str) ? '' : substr($str, 0, -2);
+        return empty($names) ? '' : $str;
```

Comment:
Is there any reason why this can't be `empty($names) ? '' : implode(", ",$names)`?  Alternatively, what happens when `implode` is called on an empty array - if it's just `''`, could this ternary be eliminated?